src/FulfilledPromise.php

```diff
@@ -13,7 +13,7 @@ class FulfilledPromise implements PromiseInterface
 
     public function __construct($value)
     {
-        if (method_exists($value, 'then')) {
+        if (is_object($value) && method_exists($value, 'then')) {
```

Comment:
The changes here look valid to me. But I wonder why we don't just check `if ($value instanceof PromiseInterface)`? 

@mtdowling It seems the library was intentionally coded to work with some other promises. See for example also https://github.com/guzzle/promises/commit/1ef15a74fa52555067004ff296273d08674dc397
Michael, do you remember what other promises are meant to be supported? Was this before the introduction of the interface? Or what's the use-case? Can we remove this maybe legacy behavior?

Reply:
IIRC, this is based on JS implementations and how promise libraries worked across implementations. This check allowed Guzzle promises to interop with React promises for example.

Reply:
I see. I would think it's better to explicitly check `instanceof` for other interfaces or implemenations like https://github.com/reactphp/promise/blob/master/src/PromiseInterface.php
Just because there is a `then` method might not even mean the same thing or might not accept the same parameters at all.
But probably not worth to change this.